src/providers/create/ProvidersCreatePage.tsx

```diff
@@ -0,0 +1,178 @@
+    setNewSecret(newValue);
+  };
+
+  const onNewProviderChange = (newValue: V1beta1Provider) => {
+    setApiError(null);
+    setValidationError(validateProviderAndSecret(newValue, newSecret, providerNames));
+    // Sync secret with new URL
+    const updatedSecret = {
+      ...newSecret,
+      data: { ...newSecret.data, url: encode(newValue?.spec?.url ?? '') },
+    };
+    setNewProvider(newValue);
+    setNewSecret(updatedSecret);
+  };
+
+  const onProjectNameChange = (newValue: string) => {
+    setApiError(null);
+    setValidationError(validateProjectName(newValue));
+    setProjectName(newValue);
+  };
+
+  const onUpdate = async () => {
+    let secret: IoK8sApiCoreV1Secret | undefined;
+    let provider: V1beta1Provider | undefined;
+
+    toggleIsLoading();
+
+    try {
+      secret = await createProviderSecret(newProvider, {
+        ...newSecret,
+        ...(projectName && {
+          metadata: { ...newSecret.metadata, namespace: projectName },
+        }),
+      });
+    } catch (err) {
+      setApiError(err as Error | null);
+      toggleIsLoading();
+      return;
+    }
+
+    try {
+      provider = await createProvider(
+        {
+          ...newProvider,
+          ...(projectName && {
+            metadata: { ...newProvider.metadata, namespace: projectName },
+          }),
+        },
+        secret!,
+      );
+    } catch (err) {
+      setApiError(err as Error | null);
+      toggleIsLoading();
+      return;
+    }
+
+    // set secret ownership using provider uid
+    try {
+      await patchProviderSecretOwner(provider, secret);
+    } catch (err) {
+      setApiError(err as Error | null);
+      toggleIsLoading();
+      return;
+    }
+
+    // navigate to providers derails page
+    const providerURL = getResourceUrl({
+      name: provider?.metadata?.name,
+      namespace: provider?.metadata?.namespace,
+      reference: ProviderModelRef,
+    });
+
+    history.push(providerURL);
+  };
+
+  return (
+    <Form>
+      <PageSection variant="light">
+        <ProvidersCreatePageHeader apiError={apiError} />
+
+        <ProvidersCreateFormsSection
+          newProvider={newProvider}
+          newSecret={newSecret}
+          projectName={projectName}
+          onNewProviderChange={onNewProviderChange}
+          onNewSecretChange={onNewSecretChange}
+          onProjectNameChange={onProjectNameChange}
+          providerNames={providerNames}
+          providerNamesLoaded={providerNamesLoaded}
+        />
+
+        <Divider className="forklift-section-create-divider" />
+
+        <ProviderCreateActionsSection
+          newProvider={newProvider}
+          projectName={projectName}
+          namespace={namespace}
+          validationError={validationError}
+          isLoading={isLoading}
+          setActiveNamespace={setActiveNamespace}
+          onUpdate={onUpdate}
+        />
+      </PageSection>
+    </Form>
+  );
+};
+
+export default ProvidersCreatePage;
```

Comment:
u have about 30 lines of jsx code here and about 150 of others - it means u have too much logic in a jsx file please adjust